src/middlewared/middlewared/plugins/init_shutdown_script.py

```diff
@@ -104,33 +107,85 @@ class Config:
         if verrors:
             raise verrors
 
+    async def execute_task(self, task):
+        task_type = task['type']
+        cmd = None
+
+        if task_type == 'COMMAND':
+            cmd = task['command']
+        elif os.path.exists(task['script'] or '') and os.access(task['script'], os.X_OK):
+            cmd = f'exec {task["script"]}'
+
+        if cmd:
+            proc = await Popen(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                shell=True,
+                close_fds=True
+            )
+            await proc.wait()
```

Comment:
You cant just wait, it will block indefinitely if the pipe gets full, use communicate or redirect stdout+stderr to DEVNULL.
communicate is preferred to report stderr.